app/services/tag_hashtag_data_source.rb

```diff
@@ -55,4 +55,36 @@ def self.search(guardian, term, limit)
   def self.search_sort(search_results, _)
     search_results.sort_by { |result| result.text.downcase }
   end
+
+  def self.search_without_term(guardian, limit)
+    return [] if !SiteSetting.tagging_enabled
+
+    tags = DB.query(<<~SQL, limit: limit).map(&:name)
+    SELECT tags.name, MAX(posts.created_at)
+    FROM tags
+    INNER JOIN topic_tags ON topic_tags.tag_id = tags.id
+          INNER JOIN topics ON topic_tags.topic_id = topics.id
+          INNER JOIN posts ON topics.id = posts.topic_id
+          WHERE posts.deleted_at IS NULL AND topics.deleted_at IS NULL
+            AND posts.created_at > (NOW() - INTERVAL '2 WEEKS')
+    GROUP BY tags.name
+    ORDER BY MAX(posts.created_at) DESC
+    LIMIT :limit
+    SQL
```

Comment:
I have the same concerns as https://github.com/discourse/discourse/pull/19219/files#r1035433392 here but we don't have the same columns we can rely on for the `tags` table. Not sure if we can bring it back to product but I wonder if a simpler heuristic of using `Tag#topic_count` is acceptable.

Reply:
Good point, maybe we go by counts instead of recents for this and categories, that will be a lot simpler. Will ask internally.